### PR TITLE
JIRA: VZ-5682 Configuration for disabling istio side car injection

### PR DESCRIPTION
**Pull request by jmaron99**
# Description

- Added an "injection-enabled" boolean setting to the Istio component in VZ CR
- Code added to various controller/operators to remove or set "istio-injection" label to appropriate setting based on value in VZ CR
- Multiple tests updated to allow for istio-injection to be enabled/disabled based on command line switch (--istioInjection={enabled|disabled})
- Created pipeline jenkins file for minimal installation testing
- Added minimal installation pipeline to push triggered pipeline

Minimal install pipeline:  https://build.verrazzano.io/job/verrazzano-minimal-install-tests/job/jmaron%252FVZ-5682/

Fixes VZ-5682

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/mysql/mysql.go [hunk 1]
```diff
@@ -115,7 +115,10 @@ func preInstall(compContext spi.ComponentContext, namespace string) error {
 			ns.Labels = make(map[string]string)
 		}
 		ns.Labels["verrazzano.io/namespace"] = namespace
-		ns.Labels["istio-injection"] = "enabled"
+		istio := compContext.EffectiveCR().Spec.Components.Istio
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/mysql/mysql.go:118]: Not for this PR, but I wonder if we should leverage the ComponentContext for this.  Have a map of data or some object that we can use to store or cache info about cross-cutting concerns like this .

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go [hunk 2]
```diff
@@ -579,11 +579,18 @@ type IstioComponent struct {
 	// +optional
 	Enabled *bool `json:"enabled,omitempty"`
 	// +optional
+	InjectionEnabled *bool `json:"injection-enabled,omitempty"`
```
Comment (pfmackin) [inline platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:582]: If Enabled is false and InjectionEnabled is true will the code do the injection?

Comment (pfmackin) [inline platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:582]: Is the InjectionEnabled default true?

File: platform-operator/controllers/verrazzano/controller.go [hunk 3]
```diff
@@ -915,24 +915,34 @@ func (r *Reconciler) getInternalConfigMap(ctx context.Context, vz *installv1alph
...
 	err := r.Get(ctx, types.NamespacedName{Name: vzconst.VerrazzanoSystemNamespace}, &vzSystemNS)
 	if err != nil {
+		log.Debugf("Creating Verrazzano system namespace")
 		if !errors.IsNotFound(err) {
 			log.Errorf("Failed to get namespace %s: %v", vzconst.VerrazzanoSystemNamespace, err)
 			return err
 		}
 		vzSystemNS.Name = vzconst.VerrazzanoSystemNamespace
 		vzSystemNS.Labels, _ = mergeMaps(nil, systemNamespaceLabels)
+		log.Debugf("Creating Verrazzano system namespace. Labels: %v", vzSystemNS.Labels)
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/controller.go:937]: this should be Log.Oncef

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go (hunk 2 above)
Comment (jmaron99) [reply platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:582]: No

Comment (jmaron99) [reply platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:582]: yes

File: platform-operator/controllers/verrazzano/controller.go [hunk 4]
```diff
@@ -915,24 +915,34 @@ func (r *Reconciler) getInternalConfigMap(ctx context.Context, vz *installv1alph
...
 		if err := r.Create(ctx, &vzSystemNS); err != nil {
 			log.Errorf("Failed to create namespace %s: %v", vzconst.VerrazzanoSystemNamespace, err)
 			return err
 		}
 		return nil
 	}
 	// Namespace exists, see if we need to add the label
+	log.Debugf("Updating Verrazzano system namespace")
```
[the remark this answers is not in the document]
Comment (jmaron99) [reply platform-operator/controllers/verrazzano/controller.go:945]: I added it primarily for debugging purposes.  Do you want it as a permanent info level message?

File: platform-operator/controllers/verrazzano/controller.go (hunk 3 above)
Comment (jmaron99) [reply platform-operator/controllers/verrazzano/controller.go:937]: Again - I added it for debug purposes. Do you feel it has merit as an info level message, appearing once?

File: tests/e2e/config/scripts/install-verrazzano-kind-minimal-install.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020, 2022, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+apiVersion: install.verrazzano.io/v1alpha1
+kind: Verrazzano
```
Comment (bobdonat) [inline tests/e2e/config/scripts/install-verrazzano-kind-minimal-install.yaml:5]: this looks like a maximum install, not minimum.  I think everything is getting installed with istio disabled.

Comment (jmaron99) [reply tests/e2e/config/scripts/install-verrazzano-kind-minimal-install.yaml:5]: I'm primarily focused on the istio injection being disabled.  If you want me to change the name to something like "install-verrazzano-no-istio-injection" I can do that.

Comment (bobdonat) [reply tests/e2e/config/scripts/install-verrazzano-kind-minimal-install.yaml:5]: I think that would be bette

**Review by pfmackin — APPROVED** [review]

**Review by michael-cico — DISMISSED** [review]

**Review by michael-cico — APPROVED** [review]

**Pull request merged**